base/broadcast.jl

```diff
@@ -1242,7 +1242,31 @@ macro __dot__(x)
     esc(__dot__(x))
 end
 
-@inline broadcasted_kwsyntax(f, args...; kwargs...) = broadcasted((args...)->f(args...; kwargs...), args...)
+"""
+    Base.Broadcast.FixKwargs{F,K} <: Function
+
+An instance `c` of `FixKwargs` is a function that takes positional
+arguments.  `c(args...)` is equivalent to `c.f(args...; c.kwargs...)`.
+
+The callables of type `FixKwargs` are created when function calls with
+keyword arguments are used in the dot-call syntax.  The dot-call
+sub-expression `f.(args...; kwargs...)` is lowered to a form that is
+equivalent to `broadcasted(FixKwargs(f, kwargs), args...)`.
+
+# Properties
+- `f::F`: a callable
+- `kwargs::K`: the keyword arguments passed to `f`
+"""
+struct FixKwargs{F,K} <: Function
+    f::F
+    kwargs::K
+end
+
+FixKwargs(::Type{T}, kwargs::K) where {T,K} = FixKwargs{Type{T},K}(T, kwargs)
```

Comment:
Maybe also support a constructor with the kwargs actually being passed as kwargs?

Reply:
How does method specialization work with functions passed to the constructors here?  Do we also need a constructor that explicitly specializes on `Function`s like this?

Reply:
I don't think we need a specialization for `Function` because we automatically gets it:

```julia
julia> typeof(Base.FixKwargs(reduce; init = 0))
Base.FixKwargs{typeof(reduce),Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol},NamedTuple{(:init,),Tuple{Int64}}}}
```

(see `typeof(reduce)`)

IIUC `Type` is special here because `julia`'s type system "leaks" the representation of types rather than decoupling it from subtyping.